crypto/kzg/src/lib.rs

```diff
@@ -156,84 +166,92 @@ impl Kzg {
+            .map_err(Error::ProverKZG)?;
+
+        // Convert the proof type to a c-kzg proof type
+        let c_kzg_proof = proofs.map(KzgProof);
+        Ok((cells, c_kzg_proof))
     }
 
     /// Verifies a batch of cell-proof-commitment triplets.
     ///
     /// Here, `coordinates` correspond to the (row, col) coordinate of the cell in the extended
     /// blob "matrix". In the 1D extension, row corresponds to the blob index, and col corresponds
     /// to the data column index.
-    pub fn verify_cell_proof_batch(
+    pub fn verify_cell_proof_batch<'a>(
         &self,
-        cells: &[Cell],
+        cells: &[CellRef<'a>],
         kzg_proofs: &[Bytes48],
         coordinates: &[(u64, u64)],
         kzg_commitments: &[Bytes48],
     ) -> Result<(), Error> {
         let (rows, columns): (Vec<u64>, Vec<u64>) = coordinates.iter().cloned().unzip();
-        if !c_kzg::KzgProof::verify_cell_kzg_proof_batch(
-            kzg_commitments,
-            &rows,
-            &columns,
-            cells,
-            kzg_proofs,
-            &self.trusted_setup,
-        )? {
-            Err(Error::KzgVerificationFailed)
-        } else {
-            Ok(())
-        }
-    }
+        // The result of this is either an Ok indicating the proof passed, or an Err indicating
+        // the proof failed or something else went wrong.
+
+        let proofs: Vec<PeerDASBytes48> = kzg_proofs
+            .iter()
+            .map(|proof| proof.as_ref())
+            .collect();
+        let commitments: Vec<PeerDASBytes48> = kzg_commitments
+            .iter()
+            .map(|commitment| commitment.as_ref())
+            .collect();
+        let verification_result = self.context.verifier_ctx().verify_cell_kzg_proof_batch(
+            commitments.to_vec(),
+            rows,
+            columns,
+            cells.to_vec(),
+            proofs.to_vec(),
+        );
 
-    pub fn cells_to_blob(&self, cells: &[Cell; c_kzg::CELLS_PER_EXT_BLOB]) -> Result<Blob, Error> {
-        Ok(Blob::cells_to_blob(cells)?)
+        // Modify the result so it matches roughly what the previous method was doing.
+        match verification_result {
+            Ok(_) => Ok(()),
+            Err(VerifierError::InvalidProof) => Err(Error::KzgVerificationFailed),
+            Err(e) => Err(Error::VerifierKZG(e)),
+        }
     }
 
-    pub fn recover_all_cells(
+    pub fn recover_cells_and_compute_kzg_proofs<'a>(
         &self,
         cell_ids: &[u64],
-        cells: &[Cell],
-    ) -> Result<Box<[Cell; c_kzg::CELLS_PER_EXT_BLOB]>, Error> {
-        Ok(c_kzg::Cell::recover_all_cells(
-            cell_ids,
-            cells,
-            &self.trusted_setup,
-        )?)
+        cells: &[CellRef<'a>],
+    ) -> Result<([Cell; CELLS_PER_EXT_BLOB], [KzgProof; CELLS_PER_EXT_BLOB]), Error> {
+        let (cells, proofs) = self
+            .context
+            .prover_ctx()
+            .recover_cells_and_proofs(cell_ids.to_vec(), cells.to_vec(), vec![])
```

Comment:
We can improve on the crypto API here, and take &[Cell] instead of Vec<Cell>, same for other parameters.

proofs is left as an empty parameter because the underlying library does not use it, it just recomputes every proof each time